src/main/scala/mesosphere/marathon/health/HealthCheckActor.scala

```diff
@@ -87,11 +88,15 @@ class HealthCheckActor(
 
   def dispatchJobs(): Unit = {
     log.debug("Dispatching health check jobs to workers")
-    taskTracker.getTasks(appId).foreach { task =>
-      if (task.getVersion == appVersion && task.hasStartedAt) {
-        log.debug("Dispatching health check job for task [{}]", task.getId)
-        val worker: ActorRef = context.actorOf(workerProps)
-        worker ! HealthCheckJob(task, healthCheck)
+    appRepository.app(appId, Timestamp(appVersion)).foreach { maybeApp =>
```

Comment:
This actor is for one application/version and one health check.
Can you pass the appDefinition in the ctor?